runtime/src/main/cpp/Porting.cpp

```diff
@@ -110,16 +110,19 @@ void consolePrintf(const char* format, ...) {
   char buffer[1024];
   va_list args;
   va_start(args, format);
-  int rv = vsnprintf_impl(buffer, sizeof(buffer) - 1, format, args);
+  int rv = vsnprintf_impl(buffer, sizeof(buffer), format, args);
+  if (rv > sizeof(buffer) - 1) rv = sizeof(buffer) - 1;  // TODO: Consider realloc or report truncating.
```

Comment:
Also needs a check for `rv < 0`. Although, I'm not sure what to do in this case. `RuntimeAssert` itself calls `konan::snprintf`, and if `snprintf` failed because it failed to allocate memory…